spec.html

```diff
@@ -21,20 +21,24 @@ <h1>Object.fromEntries ( iterable )</h1>
     1. Let _obj_ be ObjectCreate(%ObjectPrototype%).
     1. Assert: _obj_ is an extensible ordinary object with no own properties.
     1. Let _iter_ be ? GetIterator(_iterable_).
-    1. Repeat,
-      1. Let _next_ be ? IteratorStep(_iter_).
-      1. If _next_ is *false*, return _obj_.
-      1. Let _nextItem_ be ? IteratorValue(_next_).
-      1. If Type(_nextItem_) is not Object, then return ? IteratorClose(_iter_, ThrowCompletion(a newly created *TypeError* object)).
-      1. Let _key_ be Get(_nextItem_, `"0"`).
-      1. If _key_ is an abrupt completion, return ? IteratorClose(_iter_, _key_).
-      1. Let _key_ be _key_.[[Value]]
-      1. Let _value_ be Get(_nextItem_, `"1"`).
-      1. If _value_ is an abrupt completion, return ? IteratorClose(_iter_, _value_).
-      1. Let _value_ be _value_.[[Value]]
-      1. Let _propertyKey_ be ToPropertyKey(_key_).
-      1. If _propertyKey_ is an abrupt completion, return ? IteratorClose(_iter_, _propertyKey_).
-      1. Let _propertyKey_ be _propertyKey_.[[Value]]
-      1. Peform ! CreateDataPropertyOrThrow(_obj_, _propertyKey_, _value_).
+    1. If _iter_ is *undefined* or *null*, return _obj_.
```

Comment:
How can this be reached? If `GetIterator` returns an abrupt completion, we will not get here. Otherwise, `iter` will always be a `Record`.

Reply:
When the hint is sync, and Symbol.iterator is undefined or null, GetIterator returns undefined or null. https://tc39.github.io/ecma262/#sec-getmethod